src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -633,8 +633,11 @@ private Conversion DeriveStandardExplicitFromOppositeStandardImplicitConversion(
                 case ConversionKind.ImplicitReference:
                     impliedExplicitConversion = Conversion.ExplicitReference;
                     break;
-                case ConversionKind.Boxing:
-                    impliedExplicitConversion = Conversion.Unboxing;
+                case ConversionKind.ValueTypeBoxing:
+                    impliedExplicitConversion = Conversion.ValueTypeBoxing;
+                    break;
+                case ConversionKind.TypeParameterBoxing:
+                    impliedExplicitConversion = Conversion.TypeParameterBoxing;
```

Comment:
>impliedExplicitConversion = Conversion.TypeParameterBoxing; [](start = 20, length = 59)

I am not sure if this is that simple, given the current implementation of ClassifyStandardImplicitConversion